### ISSUE TITLE
Build failure: scipy (test timeout)

### ISSUE THREAD

**Issue by FliegendeWurst** [NixOS/nixpkgs#357169]
## Steps To Reproduce

1. build `python312Packages.scipy`
2. observe the test timeout (at least that is what I think it is)

## Build log

<details>

<summary>Build Log</summary>

```
=========================== short test summary info ============================
       > FAILED lib/python3.12/site-packages/scipy/optimize/tests/test_milp.py::test_milp_timeout_16545[options0-Time limit reached. (HiGHS Status 13:] - assert None is not None
       > = 1 failed, 61056 passed, 2980 skipped, 251 xfailed, 14 xpassed, 154 warnings in 7310.92s (2:01:50) =
```

</details>

## Metadata

doesn't matter

## Notify maintainers

@doronbehar

---

Note for maintainers: Please tag this issue in your PR.

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

**Comment by FliegendeWurst** [comment]
Here you go:
 - system: `"x86_64-linux"`
 - host os: `Linux 6.6.60, NixOS, 24.05 (Uakari), 24.05.20241109.fd66cd1`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.18.8`
 - channels(root): `"nixos-21.05-21.05.1268.21b696caf39"`
 - channels(arne): `"nixos-21.05-21.05.1268.21b696caf39"`
 - nixpkgs: `/home/<username>/nixpkgs

Though it should only depend on the performance of the system, not on any of the above.

**Comment by doronbehar** [comment]
> * channels(me): `"nixos-21.05-21.05.1268.21b696caf39"`

Is 21b696caf39 the Nixpkgs revision you can reproduce the issue? I'm just wondering how come you don't simply take the prebuilt Scipy from Hydra...

**Comment by FliegendeWurst** [comment]
No, I have not used the channels in a long time. The failure is on https://github.com/FliegendeWurst/nixpkgs/commit/161880616a289b09725263aeccf6a54c54d2d66a, though there is no difference to nixpkgs master for this specific package.

I didn't take the prebuilt scipy since I tried enabling strictDeps. (the test timeout occurs with or without strictDeps)

**Comment by doronbehar** [comment]
> The failure is on [FliegendeWurst@1618806](https://github.com/FliegendeWurst/nixpkgs/commit/161880616a289b09725263aeccf6a54c54d2d66a), though there is no difference to nixpkgs master for this specific package.

If Nix doesn't take the prebuilt derivation from Hydra, then the derivation you try to build and the one defined at mainstream Nixpkgs are not the same. Try to run `nix derivation show .\#python312Packages.scipy` on your fork and on the Nixpkgs `master` and debug what is the difference. It may be that the failure stems from a difference in the way you compile one of the dependencies.

This bug tracker doesn't have the resources to help people who fork Nixpkgs. You need to be able to debug failures that only you experience on your own. Hence, I'm closing this issue. Feel free though to suggest changes to the scipy expression, that you think would benefit other users like you.

**Comment by FliegendeWurst** [comment]
It seems upstream is aware of the issue and disabled the test: https://github.com/scipy/scipy/pull/19255#issuecomment-1778438888

Meaning it will not be an issue in scipy > 1.14.1
